app/services/mapping_migration_service.rb

```diff
@@ -0,0 +1,90 @@
+class MappingMigrationService
+  attr_reader :integration_name, :rtp_id, :template_file, :templates_dir, :upload_integrations
+
+  def initialize
+    @templates_dir = Configuration.paths_templates_plugins
+    @upload_integrations = Dradis::Plugins::with_feature(:upload)
+  end
+
+  def call
+    upload_integrations.each do |integration|
+      @integration_name = integration.plugin_name.to_s
+
+      template_files.each do |template_file|
+        @template_file = template_file
+
+        if defined?(Dradis::Pro)
+          migrate_pro
+        else
+          migrate
+        end
+        # delete the .template files after migrating them to the db
+        File.delete(template_file)
+      end
+    end
+  end
+
+  private
+
+  def create_mapping
+    mapping_source = File.basename(template_file, '.template')
+    destination = rtp_id ? "rtp_#{rtp_id}" : nil
+
+    Mapping.find_or_create_by!(
+      component: integration_name,
+      source: mapping_source,
+      destination: destination
+    )
+  end
+
+  def create_mapping_field(mapping, field_title)
+    mapping.mapping_fields.find_or_create_by!(
+      source_field: @source_field,
+      destination_field: field_title,
+      content: @updated_content
+    )
+  end
+
+  def migrate
+    # for each file, create a mapping for the uploader&plugin_name combination
+    ActiveRecord::Base.transaction do
+      mapping = create_mapping
+
+      template_fields.each do |field_title, field_content|
+        # set source_field by taking the first match to the existing %% syntax
+        source_field = field_content.match(/%(?<field>\S*?)%/)
+        @source_field =
+          if source_field && !source_field['field'].empty?
+            source_field['field']
```

Comment:
I'm not sure if I'm missing something but I think we can use the same regex as in line 86.
```suggestion
        source_field = field_content.match(/%(\S*?)%/)
        @source_field =
          if source_field
            source_field[1]
```